### ISSUE TITLE
(OpenSearch): (Unable to update stack to reflect index changes)

### ISSUE THREAD

**Issue by mrgiba** [awslabs/generative-ai-cdk-constructs#443]
### Describe the bug

I have a OpenSearch Serverless resource defined through the GenAI CDK Constructs, including a index
If I try to change the definition of the index, say by adding one new mapping, and run `cdk deploy`, I get an error saying **Update not supported**. An additional problem is that the stack is unable to rollback the changes, apparently due to another issue, so that I am forced to delete and recreate the stack

I attached two snapshots, one for the failed update and another for the failed rollback
![update-failure](https://github.com/awslabs/generative-ai-cdk-constructs/assets/8590331/4111a6cc-e674-480c-a1c6-d8e4d818f08d)
![rollback-failure](https://github.com/awslabs/generative-ai-cdk-constructs/assets/8590331/560a5082-0363-49d4-abe4-2df3bbc5856d)

### Expected Behavior

Update to complete successfully

### Current Behavior

Update and further rollback failed, leaving the stack in a inconsistent state

### Reproduction Steps

For an existing stack, add one new mapping to a *VectorIndex* resource and run `cdk deploy`

### CDK CLI Version

2.135.0

### Node.js Version

v21.7.1

### OS

MacOS

### Language

Python

### Region experiencing the issue

us-east-1

### Code modification

.

### Service quota

- [X] I have reviewed the service quotas for this construct

**Comment by hiSandog** [comment]
If the underlying Serverless index API does not support in-place mapping updates, the construct should model the index definition as replacement/migration rather than sending an update that CloudFormation cannot roll back. A safer path is create-new-index, optionally reindex/alias, then delete the old resource only after success, with an explicit opt-in for destructive replacement. At minimum, synth-time validation or documentation should flag immutable fields before deployment so the stack is not left wedged.
